web/src/drafts.js

```diff
@@ -167,6 +167,22 @@ export function rename_stream_recipient(old_stream_id, old_topic, new_stream_id,
     }
 }
 
+export function count_same_recipient_drafts(state) {
+    const current_drafts = draft_model.get();
+    let count = 0;
+    for (const draft_id of Object.keys(current_drafts)) {
+        const draft = current_drafts[draft_id];
+        if (
+            (state.type === "stream" && util.same_stream_and_topic(draft, state)) ||
+            (state.type === "private" &&
+                state.private_message_recipient === draft.private_message_recipient)
+        ) {
+            count = count + 1;
+        }
+    }
+    return count;
+}
```

Comment:
I think we should use `filter_drafts_by_compose_box_and_recipient` for this; I proposed moving that into this module in #28731.

Reply:
I didn't realise that that was a pre-existing function, until now. Thanks!

Reply:
@evykassirer When you move `filter_drafts_by_compose_box_and_recipient` into `drafts.ts` it'll probably be a good idea to also give its 1 argument a default value of all the drafts (`drafts.draft_model.get()`) or maybe define it within the function body.

That will make it easy to export and use outside `drafts.ts`

Reply:
I merged moving that function; @evykassirer is sick this week so you should feel free to pick up doing next-step refactoring of that function to serve your needs.

Reply:
Changed to use this function instead of a new one